### PR TITLE
[DataGrid] Fix page shift after toggling column

### LINKED ISSUE #1148: [DataGrid] Clicking 'Filters' button in the toolbar shift the view to the bottom of the page
On the first click of the 'Filters' button, the users view is shifted to the bottom of the page. Depending on the page height, this can mean that the filters are no longer in view at all. 

This only occurs on the first click. On subsequent clicks the filter overly is shown without any shift in view as expected. It also does not occur when using DataGrid only XGrid  

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Clicking on the Filters button shifts the users view to the bottom of the page

## Expected Behavior 🤔

Clicking on the filters button displays the filter overlay without and shift in view

## Steps to Reproduce 🕹

You can reproduce the issue easily with this code:

````
import * as React from 'react';
import { useDemoData } from '@material-ui/x-grid-data-generator';
//import { DataGrid, GridToolbar } from '@material-ui/data-grid';
import {XGrid, GridToolbar} from "@material-ui/x-grid";

export default function BasicToolbarFilteringGrid() {
  const { data } = useDemoData({
    dataSet: 'Commodity',
    rowLength: 50,
    maxColumns: 6,
  });

  return (
    <div style={{ height: 2400, width: '100%' }}>
      <XGrid
        {...data}
        components={{
          Toolbar: GridToolbar,
        }}
      />
    </div>
  );
}
````

Steps:

1. Refresh the page
2. Click 'Filters'

You can switch the XGrid above for DataGrid to see it working as expected

## Context 🔦

It's really affecting the user experience of our site. We can't have the view jumping around when you click buttons. 

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
  Don't forget to mention which browser you used.
  Output from `npx @material-ui/envinfo` goes here.
```
</details>

```
System:
    OS: macOS 11.2.1
  Binaries:
    Node: 14.15.0 - ~/.nvm/versions/node/v14.15.0/bin/node
    Yarn: 1.22.10 - ~/dev/Rentokil/myInitial/rentokil-myinitial/node_modules/.bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.0/bin/npm
  Browsers:
    Chrome: 88.0.4324.192
    Edge: Not Found
    Firefox: Not Found
    Safari: 14.0.3
  npmPackages:
    @material-ui/core: ~4.11.0 => 4.11.2
    @material-ui/data-grid: ^4.0.0-alpha.21 => 4.0.0-alpha.21
    @material-ui/icons: ~4.9.1 => 4.9.1
    @material-ui/lab: ^4.0.0-alpha.57 => 4.0.0-alpha.57
    @material-ui/pickers: ^3.2.10 => 3.2.10
    @material-ui/styles: ^4.11.2 => 4.11.2
    @material-ui/system:  4.11.2
    @material-ui/types:  5.1.0
    @material-ui/utils:  4.11.2
    @material-ui/x-grid: ^4.0.0-alpha.21 => 4.0.0-alpha.21
    @material-ui/x-grid-data-generator: ^4.0.0-alpha.21 => 4.0.0-alpha.21
    @material-ui/x-license:  4.0.0-alpha.21
    @types/react:  17.0.0
    react: ~16.14.0 => 16.14.0
    react-dom: ~16.14.0 => 16.14.0
    typescript: ~4.0.3 => 4.0.5

```

## Order id 💳

[Order #21167] (February 9, 2021)

### LINKED ISSUE #1254: [DataGrid] GridColumnPanel column search TextField loses focus after typing single character
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
Whenever trying to type text into GridColumnPanel column search TextField, I frequently see the GridColumnPanel lose focus after typing a single character.

https://user-images.githubusercontent.com/78176335/111718205-e31b7500-882f-11eb-8ba9-3f13c272e106.mp4

Video example comes from https://material-ui.com/components/data-grid/rendering/#toolbar using Google Chrome. I've also seen the same issue occur within Firefox.

## Expected Behavior 🤔
Column search TextField should not lose focus and user should be able to continue typing without needing to reselect TextField after a single character.

## Steps to Reproduce 🕹
1. Click on "Column" button in GridToolbar
2. Type single character in column search field
3. Observe that the column search field has lost focus

v4.0.0-alpha.12

### PR DESCRIPTION
**Pull request by m4theushw**
Fix #1148
Fix #1254

### CODE REVIEW

File: packages/grid/_modules_/grid/components/panel/GridPanelWrapper.tsx [hunk 1]
```diff
@@ -17,13 +17,15 @@ const useStyles = makeStyles(
   { name: 'MuiDataGridPanelWrapper' },
 );
 
+const isEnabled = () => true;
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/panel/GridPanelWrapper.tsx:20]: this one can be moved in the upper scope. So it's not recreated at every render

Comment (oliviertassinari) [reply packages/grid/_modules_/grid/components/panel/GridPanelWrapper.tsx:20]: How is it not already the upper scope?

Comment (dtassone) [reply packages/grid/_modules_/grid/components/panel/GridPanelWrapper.tsx:20]: Ah yes I thought I saw it in the component.

File: packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx [hunk 2]
```diff
@@ -196,5 +196,29 @@ describe('<DataGrid /> - Toolbar', () => {
...
+          <DataGrid
+            {...baselineProps}
+            components={{
+              Toolbar: GridToolbar,
+            }}
+          />
+        </div>,
+      );
+
+      const button = getByRole('button', { name: 'Select columns' });
```
Comment (oliviertassinari) [inline packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx:212]: After hesitations with Sebastian, we have been leaning toward the global approach, simpler

```suggestion
      const button = screen.getByRole('button', { name: 'Select columns' });
```

**Review by oliviertassinari — APPROVED** [review]
It looks great. 

I see that you didn't mark #1254 to be closed by this pull request, is it to take care of this demo: https://next.material-ui.com/components/trap-focus/#BasicTrapFocus.tsx or because you thought that GitHub was smart enough to handle multiple fixes?

Screenshot 2021-03-25 at 12 24 38

**Comment by m4theushw** [comment]
I knew that GitHub would not close both. I didn't know what was the description convention for multiple fixes used here. 😛

From https://github.com/mui-org/material-ui-x/issues/1254#issuecomment-803343725:

> We improve the implementation of the TrapFocus in the core, the current demos in https://next.material-ui.com/components/trap-focus/ are not right. We risk losing events because we resubscribe too often to the document.

I think it's better to open another issue in material-ui to change the docs. I can do that.

**Pull request merged**